packages/node-experimental/src/sdk/api.ts

```diff
@@ -0,0 +1,225 @@
+  return context.with(context.active(), () => callback(getCurrentScope()));
+}
+
+/**
+ * Fork a scope from the current scope, and make it the current scope in the given callback.
+ * Additionally, also setup a new isolation scope for the callback.
+ */
+export function withIsolationScope<T>(callback: (scope: Scope, isolationScope: Scope) => T): T {
+  const ctx = context.active();
+  const currentScopes = getScopesFromContext(ctx);
+  const scopes = currentScopes
+    ? { ...currentScopes }
+    : {
+        scope: getCurrentScope(),
+        isolationScope: getIsolationScope(),
+      };
+
+  scopes.isolationScope = scopes.isolationScope.clone();
+
+  return context.with(setScopesOnContext(ctx, scopes), () => {
+    return callback(getCurrentScope(), getIsolationScope());
+  });
+}
+
+/** Get the ID of the last sent error event. */
+export function lastEventId(): string | undefined {
+  return getCurrentScope().lastEventId();
+}
+
+/**
+ * Configure teh current scope.
+ * @deprecated Use `getCurrentScope()` instead.
+ */
+export function configureScope(callback: (scope: Scope) => void): void {
+  callback(getCurrentScope());
+}
+
+/** Capture an exception to Sentry. */
+export function captureException(exception: unknown, hint?: ExclusiveEventHintOrCaptureContext): string {
+  return getCurrentScope().captureException(exception, parseEventHintOrCaptureContext(hint));
+}
+
+/** Capture a message to Sentry. */
+export function captureMessage(
+  message: string,
+  // eslint-disable-next-line deprecation/deprecation
+  captureContext?: CaptureContext | Severity | SeverityLevel,
+): string {
+  // This is necessary to provide explicit scopes upgrade, without changing the original
+  // arity of the `captureMessage(message, level)` method.
+  const level = typeof captureContext === 'string' ? captureContext : undefined;
+  const context = typeof captureContext !== 'string' ? { captureContext } : undefined;
+
+  return getCurrentScope().captureMessage(message, level, context);
+}
+
+/** Capture a generic event to Sentry. */
+export function captureEvent(event: Event, hint?: EventHint): string {
+  return getCurrentScope().captureEvent(event, hint);
+}
+
+/**
+ * Add a breadcrumb.
+ */
+export function addBreadcrumb(breadcrumb: Breadcrumb, hint?: BreadcrumbHint): void {
+  const client = getClient();
+
+  const { beforeBreadcrumb, maxBreadcrumbs } = client.getOptions();
+
+  if (maxBreadcrumbs && maxBreadcrumbs <= 0) return;
+
+  const timestamp = dateTimestampInSeconds();
+  const mergedBreadcrumb = { timestamp, ...breadcrumb };
+  const finalBreadcrumb = beforeBreadcrumb
+    ? (consoleSandbox(() => beforeBreadcrumb(mergedBreadcrumb, hint)) as Breadcrumb | null)
+    : mergedBreadcrumb;
+
+  if (finalBreadcrumb === null) return;
+
+  if (client.emit) {
+    client.emit('beforeAddBreadcrumb', finalBreadcrumb, hint);
+  }
+
+  getIsolationScope().addBreadcrumb(finalBreadcrumb, maxBreadcrumbs);
```

Comment:
h: Something about this strikes me as confusing. What if there is no isolation scope?